### PR TITLE
Improve custom label format

### PR DESCRIPTION
**Pull request by xZise**
With #143 a label for the page number and total page count was added. This pull request improves a few things based on top of that change:

1. Added selection rectangle around the custom label.
2. It does not recreate the brush/font/format instances every time and only does this once.
3. Remove the text height from the available height, when determining the scale factor. Previously it justed removed the text height, when the image was higher than wide, but in certain cases (e.g. square images) this would push the text to low.

### CODE REVIEW

File: NAPS2.Lib.WinForms/EtoForms/WinForms/WinFormsListView.cs [hunk 1]
```diff
@@ -77,45 +89,51 @@ public WinFormsListView(ListViewBehavior<T> behavior)
...
-            e.Graphics.DrawString(drawString, drawFont, drawBrush, x1, y1, drawFormat);
+            RectangleF labelRect = new(x1, y1, 0, textSize.Height);
+            float maxLabelWidth = Math.Min(textSize.Width, e.Bounds.Width - 2 * TextPadding);
+            labelRect.Inflate(maxLabelWidth / 2, 0);
 
+            if (e.Item.Selected)
+            {
+                Rectangle selectionRect = Rectangle.Truncate(labelRect);
+                selectionRect.Inflate(2, 2);
+                e.Graphics.FillRectangle(SelectionBrush, selectionRect);
```
[the remark this answers is not in the document]
Comment (xZise) [reply NAPS2.Lib.WinForms/EtoForms/WinForms/WinFormsListView.cs:119]: The reasoning behind the old look was that I tired to replicate the "original" label look as if it weren't drawing them manually. When I'm forcing it to enable labels with a dummy text it looks something like this:
grafik

Anyway I'll look into it looking more like your suggestion.

File: NAPS2.Lib.WinForms/EtoForms/WinForms/WinFormsListView.cs [hunk 2]
```diff
@@ -75,38 +79,49 @@ public WinFormsListView(ListViewBehavior<T> behavior)
...
+        SizeF textSize = SizeF.Empty;
+        int textOffset = 0;
+        if (_behavior.ShowPageNumbers)
         {
-            if (_behavior.ShowPageNumbers)
-                height = ImageSize - 12;
-            else
-                height = ImageSize;
-            width = (int) Math.Round(height * (image.Width / (double) image.Height));
+            label = (e.ItemIndex + 1).ToString() + " / " + _view.Items.Count.ToString();
```
Comment (xZise) [inline NAPS2.Lib.WinForms/EtoForms/WinForms/WinFormsListView.cs:88]: While rebasing the changes to this got overwritten. Will replace it with the string interpolation.

File: NAPS2.Lib.WinForms/EtoForms/WinForms/WinFormsListView.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (xZise) [reply NAPS2.Lib.WinForms/EtoForms/WinForms/WinFormsListView.cs:119]: Okay I've taken a closer look at the appearance in the explorer and saw that label and image are selected in one large rectangle. Thus I've now two suggestions:
Screenshot 2023-07-16 115256
Screenshot 2023-07-16 115202

Do you have a preference? I personally prefer the second design.

[the remark this answers is not in the document]
Comment (cyanfish) [reply NAPS2.Lib.WinForms/EtoForms/WinForms/WinFormsListView.cs:119]: Nice, I agree, I like the second one better too.

**Comment by cyanfish** [comment]
Sorry I haven't gotten back to this till now. I like how this looks, just a couple things:
- There are a couple changes in the master branch you'll need to merge, but those are mostly overridden by your changes.
- If ShowPageNumbers is false it should keep the old look/behavior.

**Pull request merged**